ksqldb-engine/src/test/java/io/confluent/ksql/logging/query/QueryLoggerTest.java

```diff
@@ -72,21 +77,44 @@ public void anonymizesQueriesForAllLogLevels() {
   }
 
   @Test
-  public void shouldNotLogIfQueryCannotBeParsed() {
+  public void shouldNotLogQueryIfQueryCannotBeParsed() {
     String message = " I love cats";
     String query = "CREATE CAT;";
 
     QueryLogger.debug(message, query);
     QueryLogger.error(message, query);
     QueryLogger.info(message, query);
     QueryLogger.warn(message, query);
-    testAppender
-        .getLog()
+    final List<LoggingEvent> events = testAppender.getLog();
+    events
         .forEach(
             (e) -> {
               final QueryLoggerMessage msg = (QueryLoggerMessage) e.getMessage();
-              assertNotEquals(msg.getMessage(), message);
+              assertEquals(msg.getMessage(), message);
               assertNotEquals(msg.getQuery(), query);
+              assertEquals(msg.getQuery(), "<unparsable query>");
+            });
+  }
+
+  @Test
+  public void shouldLogQueryIfQueryCannotBeParsedIfAnonymizerIsDisabled() {
```

Comment:
Do we have a test which shows that the query is anonymized otherwise?

Reply:
Yep, there is `shouldNotLogQueryIfQueryCannotBeParsed`